src/vaults/SectorCrossVault.sol

```diff
@@ -46,54 +42,57 @@ contract SectorCrossVault is SectorBase {
 					Cross Vault Interface
 	/////////////////////////////////////////////////////*/
 
-	function depositIntoVaults(Request[] calldata vaults) public onlyRole(MANAGER) {
+	function depositIntoXVaults(Request[] calldata vaults) public onlyRole(MANAGER) {
 		for (uint256 i = 0; i < vaults.length; ) {
 			address vaultAddr = vaults[i].vaultAddr;
 			uint256 amount = vaults[i].amount;
 
 			Vault memory vault = checkVault(vaultAddr);
+			if (vault.chainId == chainId) revert SameChainOperation();
 
-			if (vault.chainId == chainId) {
-				ERC20(underlying()).approve(vaultAddr, amount);
-				BatchedWithdraw(vaultAddr).deposit(amount, address(this));
-			} else {
-				_sendMessage(
-					vaultAddr,
-					vault,
-					Message(amount, address(this), address(0), chainId),
-					messageType.DEPOSIT
-				);
-
-				emit BridgeAsset(chainId, vault.chainId, amount);
-			}
+			totalChildHoldings += amount;
+			beforeWithdraw(amount, 0);
+
+			_sendMessage(
+				vaultAddr,
+				vault,
+				Message(amount, address(this), address(0), chainId),
```

Comment:
TODO: pass the fee amount in as a param, validate and subtract from amount in message

Reply:
What validate means in this context?